lib/ai_bot/playground.rb

```diff
@@ -184,25 +206,153 @@ def title_playground(post)
           end
       end
 
-      def reply_to(post)
-        reply = +""
-        start = Time.now
+      def chat_context(message, channel, persona_user)
+        has_vision = bot.persona.class.vision_enabled
 
-        post_type =
-          post.post_type == Post.types[:whisper] ? Post.types[:whisper] : Post.types[:regular]
+        if !message.thread_id
+          hash = { type: :user, content: message.message }
+          hash[:upload_ids] = message.uploads.map(&:id) if has_vision && message.uploads.present?
+          return [hash]
+        end
+
+        # I would like to use a guardian and ChatSDK::Thread.messages
+        # however membership for persona_user is far in future and I need to
+        # unconditionally grab last N messages, current SDK does not support that
+
+        #guardian = Guardian.new(persona_user)
+        #thread_messages =
+        #  ChatSDK::Thread.messages(
+        #    channel_id: channel.id,
+        #    thread_id: message.thread_id,
+        #    guardian: guardian,
+        #  )
+
+        max_messages = 40
+        if bot.persona.class.respond_to?(:max_context_posts)
+          max_messages = bot.persona.class.max_context_posts || 40
+        end
```

Comment:
```suggestion
        max_messages = bot.persona.class&.max_context_posts || 40
```

Reply:
no worky for cases like this, though our internal design can be improved `"test"&.not_there` errors